src/main/java/com/wwi21sebgroup5/cinema/services/SeatTypeService.java

```diff
@@ -14,10 +15,16 @@ public class SeatTypeService {
     public static final String REGULAR_TYPE = "REGULAR";
     public static final String LOGE_TYPE = "LOGE";
 
+    private static final HashMap<String, SeatType> seatTypes = new HashMap<>();
 
     @Autowired
     SeatTypeRepository seatTypeRepository;
 
+    public SeatType getSeatType(String seatType) {
+        return seatTypes.computeIfAbsent(seatType,
+                (key) -> this.getByName(key).get());
```

Comment:
Was ist der Punkt der hashmap? Wieso nicht einfach immer per getByName den SeatType direkt anfordern? Und würde hier nicht "null" an der Stelle des Keys in die Hashmap gespeichert werden, wenn getByName ein leeres Optional liefert?
Ich verstehe den Zweck nicht ganz, bin aber wahrscheinlich nur verwirrt gerade xD

Reply:
Dachte mir, dass das hier schöner ist, da wir uns dann jedes mal einen unnötigen Zugriff auf die Datenbank sparen. Haben im Endeffekt ja gerade nur zwei SeatTypes und sonderlich mehr werdens auch nicht. Dann würde beim ersten Zugriff der Wert hier für den Key hinterlegt werden und ab dem zweiten Mal ohne Datenbankzugriff zurückgegeben werden.

Und ja, würde null drin stehen, aber sollte ja eigentlich nicht passieren, der Aufruf funktioniert ja extra mit statischen Strings. :D